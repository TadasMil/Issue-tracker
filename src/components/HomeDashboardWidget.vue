<script setup lang="ts" name="HomeDashboardWidget">
import { computed } from 'vue'

const { title, value, color } = defineProps<{
  title: string
  value: number
  color: string
}>()

const bgColor = computed(() => `bg-${color}`)
</script>

<template>
  <b-card
    class="d-flex flex-column justify-content-center text-white text-center shadow rounded-0 border-0"
    :class="[bgColor]"
  >
    <p class="text-capitalize h5 mb-4">{{ title }}</p>
    <p class="h1">{{ value }}</p>
  </b-card>
</template>
