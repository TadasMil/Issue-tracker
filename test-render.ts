import type { RenderOptions } from '@testing-library/vue'
import BootstrapVue3 from 'bootstrap-vue-3'
import { render } from '@testing-library/vue'

const customRender = (ui: any, options?: RenderOptions) =>
  render(ui, {
    ...options,
    global: {
      ...options?.global,
      plugins: [BootstrapVue3, ...(options?.global?.plugins || [])],
    },
  })

export * from '@testing-library/vue'
export { customRender as render }
