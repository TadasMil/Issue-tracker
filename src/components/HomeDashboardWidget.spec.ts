import { describe, it, expect } from 'vitest'
import { render } from '../../test-render'

import HomeDashboardWidget from './HomeDashboardWidget.vue'

describe('HomeDashboardWidget', () => {
  it('should render the component', () => {
    const { html } = render(HomeDashboardWidget, {
      props: {
        title: 'My Title',
        value: 123,
        color: 'info',
      },
    })

    expect(html()).toMatchSnapshot()
  })
})
