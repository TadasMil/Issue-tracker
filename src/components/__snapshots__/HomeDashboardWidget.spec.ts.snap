// Vitest Snapshot v1

exports[`HomeDashboardWidget > should render the component 1`] = `
"<div class=\\"card d-flex flex-column justify-content-center text-white text-center shadow rounded-0 border-0 bg-info\\">
  <!---->
  <!---->
  <div class=\\"card-body\\">
    <!---->
    <!---->
    <p class=\\"text-capitalize h5 mb-4\\">My Title</p>
    <p class=\\"h1\\">123</p>
  </div>
  <!---->
  <!---->
</div>"
`;
